tests/unit/utils/test_fs.py

```diff
@@ -94,6 +94,15 @@ def base_path_is_symlink(path):
         ):
             self.assertFalse(contains_symlink_up_to(target_path, base_path))
 
+    @patch.object(System, "is_symlink", return_value=True)
```

Comment:
Actually, this one is not desirable here, as it will make `contains_symlink_up_to` not reach the last if.
```suggestion
```